src/ports/mips64/hardware_timer.rs

```diff
@@ -1,9 +1,355 @@
 use crate::timer::TickType;
+use core::sync::atomic::{AtomicBool, Ordering};
+use core::time::Duration;
+
+// Declare timer_tests file as child file to test private functions.
+#[cfg(test)]
+#[path = "../../../tests/mips64/timer_tests.rs"]
```

Comment:
Is it normal to do tests like this? I saw some projects, where tests are in the same file as source code. Could you make a review on this question?

Reply:
Indeed, in [rust-lang-book](https://doc.rust-lang.org/book/ch11-03-test-organization.html#testing-private-functions) shows exactly that, I also found projects [1](https://github.com/sigp/lighthouse/blob/stable/account_manager/src/validator/exit.rs#L368) and [2](https://github.com/rerun-io/rerun/blob/main/crates/store/re_chunk/src/transport.rs#L668), where tests for private functions are in the same file. But there is also a project [3](https://github.com/leptos-rs/leptos/blob/main/leptos_config/src/lib.rs#L382-L383) where they do the same as me.
I think the tests in a separate file look neater, but I can do it differently. Another suggestion I can make to make things look even neater is to move these tests to "src/ports/mips64".

Reply:
Okay, I also like it when the tests are separate.